service/gcs/bridge/bridge_unit_test.go

```diff
@@ -0,0 +1,619 @@
+	"github.com/Microsoft/opengcs/service/gcs/prot"
+	"github.com/Microsoft/opengcs/service/gcs/transport"
+	"github.com/pkg/errors"
+)
+
+func Test_Bridge_Mux_New(t *testing.T) {
+	m := NewBridgeMux()
+	if m == nil {
+		t.Error("Failed to create bridge mux")
+	}
+}
+
+func Test_Bridge_Mux_New_LazyMap(t *testing.T) {
+	m := NewBridgeMux()
+	if m.m != nil {
+		t.Error("Bridge mux non lazy map instantiation")
+	}
+}
+
+type thandler struct {
+	set bool
+}
+
+func (h *thandler) ServeMsg(w ResponseWriter, r *Request) {
+	h.set = true
+}
+
+func TestBridgeMux_Handle_NilHandler_Panic(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("The code did not panic on nil handler")
+		}
+	}()
+
+	m := NewBridgeMux()
+	m.Handle(prot.ComputeSystemCreateV1, nil)
+}
+
+func Test_Bridge_Mux_Handle_Succeeds(t *testing.T) {
+	th := &thandler{}
+	m := NewBridgeMux()
+	m.Handle(prot.ComputeSystemCreateV1, th)
+
+	var hOut Handler
+	var ok bool
+	if hOut, ok = m.m[prot.ComputeSystemCreateV1]; !ok {
+		t.Error("The handler was not successfully added.")
+	}
+
+	// Is it the correct handler?
+	hOut.ServeMsg(nil, nil)
+
+	if !th.set {
+		t.Error("The handler added was not the same handler.")
+	}
+}
+
+func TestBridgeMux_HandleFunc_NilHandleFunc_Panic(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("The code did not panic on nil handler")
+		}
+	}()
+
+	m := NewBridgeMux()
+	m.HandleFunc(prot.ComputeSystemCreateV1, nil)
+}
+
+func Test_Bridge_Mux_HandleFunc_Succeeds(t *testing.T) {
+	var set bool
+	hIn := func(ResponseWriter, *Request) {
+		set = true
+	}
+
+	m := NewBridgeMux()
+	m.HandleFunc(prot.ComputeSystemCreateV1, hIn)
+
+	var hOut Handler
+	var ok bool
+	if hOut, ok = m.m[prot.ComputeSystemCreateV1]; !ok {
+		t.Error("The handler was not successfully added.")
+	}
+
+	// Is it the correct handler?
+	hOut.ServeMsg(nil, nil)
+
+	if !set {
+		t.Error("The handler added was not the same handler.")
+	}
+}
+
+func Test_Bridge_Mux_Handler_NilRequest_Panic(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("The code did not panic on nil request to handler")
+		}
+	}()
+
+	var set bool
+	hIn := func(ResponseWriter, *Request) {
+		set = true
+	}
+
+	m := NewBridgeMux()
+	m.HandleFunc(prot.ComputeSystemCreateV1, hIn)
+	m.Handler(nil)
+}
+
+func verifyResponseIsDeaultHandler(t *testing.T, i interface{}) {
+	if i == nil {
+		t.Error("The response is nil")
+		return
+	}
+
+	base := i.(*prot.MessageResponseBase)
+	if base.Result != int32(gcserr.HrNotImpl) {
+		t.Error("The default handler did not set a -1 error result.")
+	}
+	if len(base.ErrorRecords) != 1 {
+		t.Error("The default handler did not set an error record.")
+	}
+	if !strings.Contains(base.ErrorRecords[0].Message, "bridge: function not supported") {
+		t.Error("The default handler did not return the not supported message")
+	}
+}
+func Test_Bridge_Mux_Handler_NotAdded_Default(t *testing.T) {
+	//
```

Comment:
nit: I think the pattern of having empty // lines before and after a comment is pretty uncommon in Go (at least from what I've seen).